src/HTMLTableBase.php

```diff
@@ -98,14 +92,15 @@ public function appendHeader(HTMLTableHeader $header_object, $allow_super_header
         return $header_object;
     }
 
-
     /**
      * Internal test to see if we can add an header. For instance, we can only add a super header
      * to a table if there is no group defined. And we can only create a sub Header to a group if
      * it contains no row
+     *
+     * Does not actually add the header.
+     * @return void
      **/
-    abstract public function tryAddHeader();
-
+    abstract public function tryAddHeader(): void;
```

Comment:
Not sure for type hinting changes. I have no idea if it's overridden/used; but all that "HTML*" stuff should be removed; we should make less possible changes.